src/librbd/Journal.cc

```diff
@@ -894,18 +894,23 @@ uint64_t Journal<I>::append_io_events(journal::EventType event_type,
 
   Futures futures;
   uint64_t tid;
+  uint64_t tag_tid;
   {
-    Mutex::Locker locker(m_lock);
-    assert(m_state == STATE_READY);
-
-    Mutex::Locker event_locker(m_event_lock);
-    tid = ++m_event_tid;
-    assert(tid != 0);
+    {
+      Mutex::Locker locker(m_lock);
+      assert(m_state == STATE_READY);
+      tag_tid = m_tag_tid;
+
+      Mutex::Locker event_locker(m_event_lock);
```

Comment:
Maybe move ```m_event_tid``` under the protection of ```m_lock``` so you don't need to acquire/release/re-acquire ```m_event_lock```